### PR TITLE
Implement trend parser and display trends in json form on servlet

### PR DESCRIPTION
**Pull request by alexcn3**
sorry for the code dump! This will be the first and the last!!
Screenshot 2020-06-25 at 11 53 58 PM

### CODE REVIEW

File: project/src/main/java/com/google/sps/data/TrendRSS.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  public void setTitle(String title) {
+    this.title = title;
+  }
+
+  public String getFreq() {
+    return trendFrequency;
+  }
+  
+  public void setFreq(String frequency){
```
Comment (niuamy) [inline project/src/main/java/com/google/sps/data/TrendRSS.java:33]: nit: space between closing parenthesis and opening bracket

File: project/src/main/java/com/google/sps/data/TrendRSSFeed.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  public TrendRSSFeed(String title, String trendFrequency) {
+    this.title = title;
+    this.trendFrequency = trendFrequency;
+  }
+
+  public List<TrendRSS> getTrends() {
+    return entries;
+  }
+
+  public String getTitle(){
```
Comment (niuamy) [inline project/src/main/java/com/google/sps/data/TrendRSSFeed.java:35]: nit: space between closing parenthesis and opening bracket

File: project/src/main/java/com/google/sps/data/Trend.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/* stores one instance of a trend */
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/data/Trend.java:17]: nit: capitalize Stores and end with a .

File: project/src/main/java/com/google/sps/data/TrendFrequencyParser.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Characters;
+import javax.xml.stream.events.XMLEvent;
+
+import com.google.sps.data.TrendRSS;
+import com.google.sps.data.TrendRSSFeed;
+
+public class TrendFrequencyParser {
+  //these are the xml tags that data is being extracted from
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/data/TrendFrequencyParser.java:32]: nit: space after // and turn into sentence (These... from.)

File: project/src/main/java/com/google/sps/data/TrendFrequencyParser.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          }
+        }
+      }
+    } catch (XMLStreamException e) {
+      throw new RuntimeException(e);
+    }
+    return feed;
+  }
+
+  /* used to get text from an xml tag */
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/data/TrendFrequencyParser.java:93]: same as other comments

File: project/src/main/java/com/google/sps/data/TrendRSS.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/* stores one instance of a trend/item in the RSS Feed */
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/data/TrendRSS.java:17]: same

File: project/src/main/java/com/google/sps/data/TrendRSSFeed.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/* stores the entire Google Trends RSS Feed*/
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/data/TrendRSSFeed.java:20]: same

File: project/src/main/java/com/google/sps/servlets/TrendServlet.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: make getTrends() an automated function. Right now, whenever you make a GET request to '/trends' it adds
+    // trends to the Datastore and it should only be doing this at certain times.
+    getTrends();
+    showTrends(response);
+  }
+
+  /* displays trends in json format */
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/servlets/TrendServlet.java:48]: same

File: project/src/main/java/com/google/sps/servlets/TrendServlet.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+      trends.add(trend);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(trends));
+  }
+
+  public void storeTrend(String topic, String frequency) {
```
Comment (ashley3) [inline project/src/main/java/com/google/sps/servlets/TrendServlet.java:72]: Add comments to public methods here and below

**Pull request merged**